### PR TITLE
[WIP]Use reverse Poland notation as default expression of AggrDefinitionParser

### PR DESCRIPTION
**Pull request by Renkai**
### What problem does this PR solve?
A step of https://github.com/tikv/tikv/issues/6281 , generally make as rpn expression as the default expression for all coprocessors.

### Check List 

Tests 

- Unit test

### CODE REVIEW

File: components/tidb_query_vec_expr/src/types/expr.rs [hunk 1]
```diff
@@ -2,11 +2,39 @@
...
 use std::any::Any;
 
-use tipb::FieldType;
+use tipb::{ExprType, FieldType};
 
 use super::super::function::RpnFnMeta;
 use tidb_query_datatype::codec::data_type::ScalarValue;
 
+#[derive(Debug, Clone, Copy)]
+pub enum Operator {
```
Comment (breeswish) [inline components/tidb_query_vec_expr/src/types/expr.rs:11]: What's the purpose of introducing a new `Operator` type?

File: components/tidb_query_vec_expr/src/types/expr.rs [hunk 2]
```diff
@@ -25,24 +53,29 @@ pub enum RpnExpressionNode {
     },
 
     /// Represents a reference to a column in the columns specified in evaluation.
-    ColumnRef { offset: usize },
+    ColumnRef {
+        offset: usize,
+    },
+    // Represents a operator call
+    Operator {
```
Comment (breeswish) [inline components/tidb_query_vec_expr/src/types/expr.rs:60]: Aggregations cannot be nested and must be the most outside operator, so there is no need to introduce a new variant for the Rpn node

File: components/tidb_query_vec_expr/src/types/expr.rs (hunk 1 above)
Comment (Renkai) [reply components/tidb_query_vec_expr/src/types/expr.rs:11]: > What's the purpose of introducing a new `Operator` type?

If there is no Operator type, rpn-expression can't express a full Aggregation.

File: components/tidb_query_vec_expr/src/types/expr.rs (hunk 2 above)
Comment (Renkai) [reply components/tidb_query_vec_expr/src/types/expr.rs:60]: ditto

**Comment by Renkai** [comment]
new PR at https://github.com/tikv/tikv/pull/8238

**Pull request closed without merging**